### PR TITLE
added firewall resource, changes syntax and linting

### PR DESCRIPTION
**Pull request by cjohannsen81**
added vmhost_firewall resource, changed the description of a rule to correct rubocop syntax, aligned return value if method is missing

### CODE REVIEW

**Comment by cjohannsen81** [comment]
Example output: 

```
  ✖  vmware-2.2: Ensure the AllIp property of all network service firewall exceptions is equals to false. (42 failed)
     ✖  vmhost_firewall CIMHttpServer all_ip should eq false

     expected: false
          got: true

     (compared using ==)

     ✖  vmhost_firewall CIMHttpsServer all_ip should eq false

     expected: false
          got: true

     (compared using ==)

     ✖  vmhost_firewall CIMSLP all_ip should eq false
```

File: libraries/virtual_portgroup.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class VmWareVirtualPortgroup < Inspec.resource(1)
   "
 
   example "
-    describe virtual_portgroup({datacenter: 'ha-datacenter', host: 'localhost', portgroup: 'VM Network'}) do
+    describe virtual_portgroup(datacenter: 'ha-datacenter', host: 'localhost', portgroup: 'VM Network') do
```
Comment (chris-rock) [inline libraries/virtual_portgroup.rb:12]: 👍

File: libraries/vm_advancedsetting.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ def initialize(opts)
 
   # Expose all parameters
   def method_missing(name) # rubocop:disable Style/MethodMissing
-    advancedsetting[name.to_s]
+    [name.to_s]
```
Comment (chris-rock) [inline libraries/vm_advancedsetting.rb]: any reason for that change?

File: libraries/vm_device.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def initialize(opts)
 
   # Expose all parameters
   def method_missing(name) # rubocop:disable Style/MethodMissing
-    connected[name.to_s]
```
Comment (chris-rock) [inline libraries/vm_device.rb:24]: We should call the method explicitly

File: libraries/vmhost_buildnumber.rb [hunk 4]
```diff
@@ -21,7 +21,7 @@ def initialize(opts)
 
   # Expose all parameters
   def method_missing(name) # rubocop:disable Style/MethodMissing
-    build[name.to_s]
+    [name.to_s]
```
Comment (chris-rock) [inline libraries/vmhost_buildnumber.rb]: we should call the method explicitly

File: libraries/vmhost_firewall.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  "
+
+  # Load the configuration file on initialization
+  def initialize(opts)
+    @opts = opts
+  end
+
+  # Expose all parameters
+  def method_missing(name) # rubocop:disable Style/MethodMissing
+    [name.to_s]
```
Comment (chris-rock) [inline libraries/vmhost_firewall.rb]: as described above

File: libraries/vmhost_vswitch.rb [hunk 6]
```diff
@@ -21,7 +21,7 @@ def initialize(opts)
 
   # Expose all parameters
   def method_missing(name) # rubocop:disable Style/MethodMissing
-    forgedTransmits[name.to_s]
```
Comment (chris-rock) [inline libraries/vmhost_vswitch.rb:24]: see above

Comment (cjohannsen81) [reply libraries/vmhost_vswitch.rb:24]: we have more than one method.

**Comment by cjohannsen81** [comment]
Changed it to removal of method_missing since error handling shows the wrong method name:

```
  ✖  vmware-2.2: Ensure the AllIp property of all network service firewall exceptions is equals to false. (44 failed)
     ✖  vmhost_firewall DHCPv6 tralala 
     undefined method `tralala' for vmhost_firewall DHCPv6:#<Class:0x007fc13fb0e200>
     ✖  vmhost_firewall CIMHttpServer tralala 
     undefined method `tralala' for vmhost_firewall CIMHttpServer:#<Class:0x007fc13fb0e200>
```

**Comment by chris-rock** [comment]
Awesome, Thank you @cjohannsen81

**Pull request merged**